lib/spack/spack/test/cray_manifest.py

```diff
@@ -423,3 +461,30 @@ def test_convert_validation_error(tmpdir, mutable_config, mock_packages, mutable
     with pytest.raises(cray_manifest.ManifestValidationError) as e:
         cray_manifest.read(invalid_schema_path, True)
     str(e)
+
+
+@pytest.fixture
+def directory_with_manifest(tmpdir, manifest_content):
+    """Create a manifest file in a directory. Used by 'spack external'."""
+    with tmpdir.as_cwd():
+        test_db_fname = "external-db.json"
+        with open(test_db_fname, "w") as db_file:
+            json.dump(manifest_content, db_file)
+
+    yield str(tmpdir)
+
+
+external = SpackCommand("external")
+
+
+def test_find_external_nonempty_default_manifest_dir(
+    mutable_database, mutable_mock_repo, tmpdir, monkeypatch, directory_with_manifest
+):
+    """The user runs 'spack external find'; the default manifest directory
+    contains a manifest file. Ensure that the specs are read.
+    """
+    monkeypatch.setenv("PATH", "")
+    monkeypatch.setattr(spack.cray_manifest, "default_path", str(directory_with_manifest))
+    external("find")
```

Comment:
Can we test just `_collect_and_consume_cray_manifest_file` instead of the whole `external find` command? Would we miss anything substantial that is not covered by other tests too if we do so?

Reply:
> Would we miss anything substantial that is not covered by other tests too if we do so?

No. I have changed it. I will say though that the benefit of using `SpackCommand` is that we don't have to ask questions like these in the first place: we are running within the same context that the user would.